### ISSUE TITLE
Sorting of grouped columns

### ISSUE THREAD

**Issue by maraisr** [ceolter/ag-grid#622]
So when you specify a `groupKeys`, there is no ability to sort it. Basically, I want a "group" of records sorted alphabetically.. So that all records under that group fall under the same sort. 

Maybe - is this just something to do with my datasource, where I simply just sort my array before I give it to agGrid?

Edit:
Just made it more clear what i'm trying to articulate.

**Comment by ceolter** [comment]
Sorry, I don't follow what you mean by 'all records under that group fall under the same sort'. Maybe explain an example using the test drive data???

Re proving the data sorted - yes you could do that - but that's not what I intended. The grid should sort.

You can provide your own comparator to the colDef. So if you provide a comparator to the grouping colDel (just make sure your not using the grids default provided colDef) then you can do the sorting whatever way you want when the use clicks the group header.

**Comment by maraisr** [comment]
@ceolter Cheers mate for getting back to me so promptly! After some time I managed to solve the issue I was having, simply by sorting the array of data before I was to hand if off to agGrid. 

I guess what I was trying to get at is, the grouped data, wasnt "grouping" them all at once, so you'd have group A, with 5 rows on page 1, then 5 rows of group B on page 1, then 5 rows of Group A on page, rather than have all Group A rows on page 1.. Hope that makes sense..

But when I sorted the array of data, and then apply the grouping, it'll have worked, because all the rows of Group A were together.

Also thank you so much for exposing the prototypes of the Grid, I was able to implement, dropdown "dynamic" combos on the keyboard navigation events. So the focus of a cell, Id inject a "custom" dropdown,pre populated with its "selected" item, and then a keyboard navigation out. All native (no angular). All still using the ValueGetter. 

Loved the Datasource idea, I was able to assign that to a closure, to run my API endpoint and do sorting locally... Returning 10,000 records from the DB, and then page locally with JavaScript slicing. I use a custom cascade combo filtering mechanism, so I couldn't use your filter model. But all in all, love this project!

One feature I'd love is a "blank" first or last row, that you can populate with data, and then that'll in essence run an `INSERT` statement to the DB. 

TL:DR; Issue solved, love the grid! xD

**Comment by ceolter** [comment]
cool, thanks for all the positive feedback. re the blank row - you can implement that yourself with a footer. just assign a different floatingCellRenderer to each column, and instead of rendering the row, have your own custom text field always displayed.

yes the plan was not to use Angular on the inside of the grid. this should make things easier for you. it also means you are not tied to Angular (even if you are an Angular person, this means you are free to hop between Angular 1 and Angular 2).
